controllers/postController.js

```diff
@@ -202,4 +203,126 @@ router.get('/:id/image', verifyToken, (req, res) => {
       res.json(err);
     });
 });
+
+// /////////////////////
+//      POLLS
+// /////////////////////
+router.get('/:id/poll', verifyToken, (req, res) => {
+  Post.findById(req.params.id)
+    .populate('author')
+    .populate({
+      path: 'media',
+      populate: {
+        path: 'poll',
+        populate: {
+          path: 'usersVoted',
+        },
+      },
+    })
+    .then((post) => {
+      if (!post.media.poll) {
+        res.status(400).json('Post does not have a poll');
+        return;
+      }
+      res.json(post.media.poll);
+    })
+    .catch((err) => {
+      res.status(500).json(err.toString());
+    });
+});
+
+router.post('/:id/poll', verifyToken, (req, res) => {
+  Post.findById(req.params.id)
+    .populate('author')
+    .then((post) => {
+      if (post.author._id.toString() !== req.user._id.toString()) {
+        res.status(403).json({});
+        return;
+      }
+
+      const validation = requestValidator(['options', 'title', 'multiSelect'], req.body);
+      if (validation.status !== 200) {
+        res.status(validation.status).json(validation.message);
+        return;
+      }
+
+      post.addMedia('poll', req.body).then((media) => {
+        res.json(media.poll);
+      }).catch((err) => {
+        res.status(500).json(err.toString());
+      });
+    }).catch((err) => {
+      res.status(500).json(err.toString());
+    });
+});
+
+router.post('/:id/poll/option', verifyToken, (req, res) => {
+  Post.findById(req.params.id)
+    .populate('author')
+    .populate({
+      path: 'media',
+      populate: {
+        path: 'poll',
+      },
+    })
+    .then((post) => {
+      if (!post.media.poll) {
+        res.status(400).json('Post does not have a poll');
+        return;
+      }
+
+      if (!post.media.poll.open && post.author._id.toString() !== req.user._id.toString()) {
+        res.status(403).json({});
+        return;
+      }
+
+      const validation = requestValidator(['option'], req.body);
+      if (validation.status !== 200) {
+        res.status(validation.status).json(validation.message);
+      }
+
+      post.media.poll.addOption(req.body.option).then((poll) => {
+        res.json(poll);
+      })
+        .catch((err) => {
+          res.status(500).json(err.toString());
+        });
+    })
+    .catch((err) => {
+      res.status(500).json(err.toString());
+    });
+});
+
+router.post('/:id/poll/vote', verifyToken, (req, res) => {
```

Comment:
Is there any way to unvote for an option?

Reply:
Lol somehow I did not think about that yeah I'll have to add that

Reply:
Updated